core/standards.md

```diff
@@ -0,0 +1,93 @@
+- Include a license at the top
+- Declare pragma version after the license
+- In almost all cases there should only be one contract per file
+
+### Contract ordering
+
+Following and expanding on [Solidity Order of layout recommendations](https://docs.soliditylang.org/en/stable/style-guide.html#order-of-layout) our contracts are generally laid out in this way:
+
+1. State variables (including wards and math)
+
+    a. Constants then immutables then storage variables. All constants and variables should have explicit visibility declared. In most cases, constants and immutables can be declared internal.
+
+    b. Wards should be storage slot `0x0`
+2. Events
+3. Getter functions (i.e. to make reading structs easier)
+
+    a. these may be placed at the end of the file if they are extensive to improve readability
+4. Constructor
+5. Modifiers
+6. Internal Math constants and functions
+7. Wards functions (i.e. `rely`/`deny`)
+8. Fileable functions
+9. Other Internal functions
+10. Other External functions
+
+### Spacing
+
+We align variables with extra whitespace. You will get PR comments if you are off by even one space.
+
+### Naming
+
+We prefer to try to have **Daiwanese** naming patterns (3 or 4 letter variables and functions) such as `guy`, `usr`, or `who` for address parameters and `wad` or `amt` for `uint` or `int` parameters (with care given to `RAY`, `RAD`, and `WAD` precision).
+
+If we are avoiding variable shadowing, we add a `_` to the end of the variable names.
+
+If we are dealing with internal variables or functions we add the `_` to the front of the name.
+
+If we need to interact with external contracts, we prefer to have a minimal static interface using the naming convention of `<ContractName>Like`. (These should be declared at the top of the contract file.)
+
+### Dependencies
+
+We try to keep our repos relatively dependency free.  We do not rely on external contract libraries (i.e. Open Zeppelin) and try on only import submodules that are absolutely necessary as this keeps the cascading updates to a minimum.
+
+### Antipatterns
+
+In order to limit contract complexity, we also tend to stay away from inheritance patterns.
+
+We believe that upgradable proxies are an antipattern.  If you need upgradability, it should be handled with modular design and permission given to the Pause Proxy. There are rare exceptions to this such as where there is a user storage mapping and high chance of needing to upgrade (e.g. cropper or charter).
+
+### Documentation
+
+Repositories should have a README file that outline the purpose, functionality and how to use the contract.  In addition to this external/public functions should have [NatSpec comments](https://docs.soliditylang.org/en/latest/style-guide.html#natspec).
+
+### Deployment
+
+Contracts should be verified on Etherscan with a license and pragma clearly set. Optimizations generally should be turned off.  The verified code should match the main branch of the source repository on Github.
```

Comment:
```suggestion
Contracts should be automatically verified on Etherscan with compiler version and license values populated. Optimizations should be turned off, with rare exception. Verified code should match the main branch of the source repository on GitHub.
```

Reply:
I'm not sure about the automatically verified piece. I think that is just spells.

Reply:
If it isn't autoverify then we can run into the issue where etherscan doesn't populate the license version (as we have seen in recent onboardings).

Up to you on this, I'm open to either way

Reply:
```suggestion
Contracts should be verified on Etherscan with compiler version and license values populated. Optimizations should be turned off, with rare exception. Verified code should match the main branch of the source repository on GitHub.
```